sql/core/src/main/scala/org/apache/spark/sql/execution/joins/BroadcastHashJoin.scala

```diff
@@ -47,75 +42,40 @@ case class BroadcastHashJoin(
     right: SparkPlan)
   extends BinaryNode with HashJoin with CodegenSupport {
```

Comment:
We could. It would be a matter of creating a local index creating node, modifying the index access path (we currently assume that the index is wrapped in a broadcast variable) and allowing `requiredChildDistribution` to return different distributions.